code/game/objects/items/weapons/holosign_creator.dm

```diff
@@ -0,0 +1,189 @@
+	desc = "A hologaphic projector that creates holographic engineering barriers."
+	icon_state = "signmaker_engi"
+	holosign_type = /obj/effect/overlay/holograph/barrier/engineering
+	creation_time = 30
+	max_signs = 6
+
+
+/obj/effect/overlay/holograph
+	icon = 'icons/effects/effects.dmi'
+	anchored = 1
+	var/holo_integrity = 1
+
+/obj/effect/overlay/holograph/attackby(obj/item/weapon/W, mob/user, params)
+	if(!W.force || (W.flags & (ABSTRACT|NOBLUDGEON)))
+		return
+	if(W.force >= 10)
+		take_damage(3, user)
+	else
+		take_damage(1, user)
+
+/obj/effect/overlay/holograph/blob_act()
+	qdel(src)
+
+/obj/effect/overlay/holograph/attack_animal(mob/living/simple_animal/M)
+	take_damage(5, M)
+
+/obj/effect/overlay/holograph/attack_alien(mob/living/carbon/alien/A)
+	take_damage(5, A)
+
+/obj/effect/overlay/holograph/attack_hand(mob/living/user)
+	take_damage(1, user)
+
+/obj/effect/overlay/holograph/mech_melee_attack(obj/mecha/M)
+	playsound(loc, 'sound/weapons/Egloves.ogg', 80, 1)
+	visible_message("<span class='danger'>[M.name] has hit [src].</span>")
+	qdel(src)
+
+/obj/effect/overlay/holograph/attack_slime(mob/living/simple_animal/slime/S)
+	if(S.is_adult)
+		take_damage(5, S)
+	else
+		take_damage(2, S)
+
+/obj/effect/overlay/holograph/proc/take_damage(amount, mob/living/user)
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.do_attack_animation(src)
+	playsound(loc, 'sound/weapons/Egloves.ogg', 80, 1)
+	user.visible_message("<span class='danger'>[user] hits [src].</span>", \
+						 "<span class='danger'>You hit [src].</span>" )
+	holo_integrity -= amount
+	if(holo_integrity <= 0)
+		qdel(src)
+
+/obj/effect/overlay/holograph/hitby(atom/movable/AM)
+	..()
+	var/tforce = 1
+	if(ismob(AM))
+		tforce = 5
+	else if(isobj(AM))
+		var/obj/item/I = AM
+		tforce = max(1, I.throwforce * 0.2)
+	playsound(loc, 'sound/weapons/Egloves.ogg', 80, 1)
+	holo_integrity -= tforce
+	if(holo_integrity <= 0)
+		qdel(src)
+
+/obj/effect/overlay/holograph/bullet_act(obj/item/projectile/P)
+	if((P.damage_type == BRUTE || P.damage_type == BURN))
+		holo_integrity -= P.damage * 0.5
+		if(holo_integrity <= 0)
+			qdel(src)
+
+/obj/effect/overlay/holograph/wetsign
+	name = "wet floor sign"
+	desc = "The words flicker as if they mean nothing."
+	icon_state = "holosign"
+
+/obj/effect/overlay/holograph/barrier
+	name = "holo barrier"
+	desc = "A short holographic barrier which can only be passed by walking."
+	icon_state = "holosign_sec"
+	pass_flags = LETPASSTHROW
+	density = 1
+	holo_integrity = 4
+
+/obj/effect/overlay/holograph/barrier/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(!density)
+		return 1
+	if(air_group || (height==0))
+		return 1
+	if(mover.pass_flags & (PASSGLASS|PASSTABLE|PASSGRILLE))
+		return 1
+	if(iscarbon(mover))
+		var/mob/living/carbon/C = mover
+		if(C.m_intent == "walk")
+			return 1
+
+/obj/effect/overlay/holograph/barrier/engineering
+	icon_state = "holosign_engi"
+
+/obj/item/weapon/caution
```

Comment:
It was in the file janisigns.dm that I deleted and replaced by holosign_creator.dm
I didn't know where else to put it.